src/BizHawk.Emulation.Cores/Consoles/Sony/PSP/PPSSPP.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using BizHawk.BizInvoke;
+using BizHawk.Common;
+using BizHawk.Emulation.Common;
+using BizHawk.Emulation.Cores.Properties;
+using BizHawk.Emulation.DiscSystem;
+
+namespace BizHawk.Emulation.Cores.Consoles.Sony.PSP
+{
+	[PortedCore(
+		name: CoreNames.PPSSPP,
+		author: "Henrik Rydgård et al",
+		portedVersion: "2025.03.09 (ecbbadd)",
+		portedUrl: "https://github.com/hrydgard/ppsspp",
+		isReleased: false)]
+	public partial class PPSSPP : IEmulator, IVideoProvider, ISoundProvider, IInputPollable, IStatable, ISettable<PPSSPP.Settings, PPSSPP.SyncSettings>
+	{
+		private static DynamicLibraryImportResolver _resolver;
+		private LibPPSSPP _libPPSSPP;
+		private readonly List<IDiscAsset> _discAssets;
+
+		[CoreConstructor(VSystemID.Raw.PSP)]
+		public PPSSPP(CoreLoadParameters<Settings, SyncSettings> lp)
+		{
+			Console.WriteLine("In PPSSP Core");
+
+			// Assigning configuration
+			_settings = lp.Settings;
+			_syncSettings = lp.SyncSettings;
+
+			DriveLightEnabled = true;
+			_discAssets = lp.Discs;
+
+			// If no discs loaded, then there's nothing to emulate
+			if (_discAssets.Count == 0) throw new InvalidOperationException("No CDs provided for emulation");
+			_isMultidisc = _discAssets.Count > 1;
+
+			_CDReadCallback = CDRead;
+			_CDSectorCountCallback = CDSectorCount;
+			_discIndex = 0;
+			foreach (var disc in _discAssets) _cdReaders.Add(new(disc.DiscData));
+
+			Console.WriteLine($"[CD] Sector count: {_discAssets[0].DiscData.Session1.LeadoutLBA}");
+			_syncSettings = lp.SyncSettings ?? new();
+			ControllerDefinition = CreateControllerDefinition(_syncSettings, _isMultidisc);
+
+			// Registering service provider
+			_serviceProvider = new(this);
+			_serviceProvider.Register<IVideoProvider>(this);
+			_serviceProvider.Register<ISoundProvider>(this);
+
+			// Loading LibPPSSPP
+			_resolver?.Dispose();
+			_resolver = new(OSTailoredCode.IsUnixHost ? "libppsspp.so" : "libppsspp.dll", hasLimitedLifetime: true);
+			_libPPSSPP = BizInvoker.GetInvoker<LibPPSSPP>(_resolver, CallingConventionAdapters.Native);
+
+			// Setting input callback
+			_inputCallback = InputCallback;
+			_libPPSSPP.SetInputCallback(_inputCallback);
+
+			// Setting CD callbacks
+			_libPPSSPP.SetCdCallbacks(_CDReadCallback, _CDSectorCountCallback);
+
+			//// Pre-loading emulator resources
+
+			// Getting compat.ini -- required to set game-specific compatibility flags
+			string resourceName = "";
+
+			resourceName = "compat.ini";
+			var compatIniData = new MemoryStream(Resources.PPSSPP_COMPAT_INI.Value).ToArray();

```

Comment:
Would this work for these?
```suggestion
			var compatIniData = Resources.PPSSPP_COMPAT_INI.Value.ReadAllBytes();
```

Reply:
unfortunately it didnt (compilation error)

Severity	Code	Description	Project	File	Line	Suppression State	Details
Error (active)	CS1061	'byte[]' does not contain a definition for 'ReadAllBytes' and no accessible extension method 'ReadAllBytes' accepting a first argument of type 'byte[]' could be found (are you missing a using directive or an assembly reference?)	BizHawk.Emulation.Cores	C:\Users\sergiom\Documents\BizHawk\src\BizHawk.Emulation.Cores\Consoles\Sony\PSP\PPSSPP.cs	71